src/App.tsx

```diff
@@ -1,24 +1,72 @@
 import React from 'react';
 import './App.scss';
+import classNames from 'classnames';
+import { Clock } from './Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    clockName: 'regular',
+  };
+
+  showClock = () => {
+    this.setState({ isClockVisible: true });
+  };
+
+  hideClock = () => {
+    this.setState({ isClockVisible: false });
+  };
+
+  renameClock = () => {
+    switch (this.state.clockName) {
+      case 'regular': {
+        this.setState({ clockName: 'fancy' });
+        break;
+      }
+
+      case 'fancy': {
+        this.setState({ clockName: 'regular' });
+        break;
+      }
+
+      default: {
+        throw new Error('Unknown name');
+      }
+    }
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <div className={classNames('App__content', `App__content--${this.state.clockName}`)}>
+          {this.state.isClockVisible && <Clock name={this.state.clockName} />}
+          <div className="App__buttons">
+            <button
+              type="button"
+              onClick={this.showClock}
+              className="App__button"
+            >
+              Show Clock
+            </button>
+            <button
+              type="button"
+              onClick={this.hideClock}
+              className="App__button"
+            >
+              Hide Clock
+            </button>
+            <button
+              type="button"
+              onClick={this.renameClock}
+              className="App__button"
+            >
+              Set new name
+            </button>
```

Comment:
If you want to improve your solution, you may change the disabled state of buttons if the clock is hidden/shown